wgpu/src/lib.rs

```diff
@@ -4739,7 +4765,42 @@ impl<'a> Drop for RenderPass<'a> {
     }
 }
 
-impl ComputePass {
+impl<'a> ComputePass<'a> {
+    /// Drops lifetime constraint to the parent command encoder, making usage of the encoder
+    /// while this pass is recorded a runtime error instead.
+    ///
+    /// Attention: As long as the compute pass has not been ended, any mutable operation on the parent
+    /// command encoder will cause a runtime error and invalidate it!
+    /// By default the lifetime constraint protects from this, but it can be useful
+    /// to handle this at runtime, for example in order to store the pass and encoder in the same
+    /// data structure.
+    ///
+    /// This operation has no effect on pass recording and is considered low overhead.
+    /// It's a safe operation, since [`CommandEncoder`] is in a locked state as long as the pass is active
+    /// regardless of here lifted lifetime constraint.
+    pub fn make_static(mut self) -> ComputePass<'static> {
+        // It's tempting to simply transmute the object, but that would be unsound
+        // since `repr(rust)` objects have no guarantees about padding or alignment
+        // even for seemingly equivalent types like `ComputePass<'a>` and `ComputePass<'static>`!
+
+        // Instead, make sure that dropping of `self` is a no-op.
```

Comment:
```suggestion
        // Instead, make sure that dropping of `self` is a no-op other than dropping its pointers.
```

This distinction is important because if we didn't want to drop `context`, this would be a classic place to use `ManuallyDrop` instead of introducing a flag, but in fact doing that would be a leak.